scripts/core/editor3/components/Editor3.jsx

```diff
@@ -0,0 +1,226 @@
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {Editor, EditorState, RichUtils, ContentState} from 'draft-js';
+
+export class Editor3 extends React.Component {
+    /**
+     * @ngdoc react component
+     * @module superdesk.core.editor3
+     * @name Editor3
+     * @param {Array} editorFormat the formating settings available for editor
+     * @param {String} value the model for editor value
+     * @param {Boolean} readOnly if true the editor is read only
+     * @param {Boolean} showToolbar if true the editor will show the toolbar
+     * @param {Function} onChange the callback executed when the editor value is changed
+     * @param {String} language the current language used for spellchecker
+     * @description Editor3 is a draft.js based editor that support customizable formatting (shortly more features will come ...)
+     */
+    constructor(props) {
+        super(props);
+        var initialContentState = ContentState.createFromText(props.value);
+        this.state = {editorState: EditorState.createWithContent(initialContentState)};
+
+        this.readOnly = props.readOnly || false;
+        this.showToolbar = props.showToolbar || false;
```

Comment:
Please use [React propTypes](https://facebook.github.io/react/docs/typechecking-with-proptypes.html) to document all props and their types. This is very helpful and offers type safety. They will also have to be documented.